src/lib/data/test.tsx

```diff
@@ -163,3 +170,115 @@ describe('useSettings', () => {
+
+    test('returns the stored alertList', async () => {
+        const storedAlertList = {
+            dateKeys: ['2022-01-04T16:30:00.000-05:00'],
+            '2022-01-04T16:30:00.000-05:00': ['2003019'],
+        };
+        jest.spyOn(window.localStorage.__proto__, 'getItem');
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        window.localStorage.__proto__.getItem = jest.fn(() => JSON.stringify(storedAlertList));
+        const { result } = await actAndFlush(() => renderHook(() => useAlertList(), { wrapper: MockProvider }));
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        expect(window.localStorage.getItem).toHaveBeenLastCalledWith('aiera:sdk:alertList');
+        expect(result.current.alertList).toEqual(storedAlertList);
+    });
+
+    test('add to alertList', async () => {
+        const initialAlertList = {
+            dateKeys: ['2022-01-04T16:30:00.000-05:00'],
+            '2022-01-04T16:30:00.000-05:00': ['2003019'],
+        };
+        jest.spyOn(window.localStorage.__proto__, 'getItem');
+        jest.spyOn(window.localStorage.__proto__, 'setItem');
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        window.localStorage.__proto__.getItem = jest.fn(() => JSON.stringify(initialAlertList));
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        window.localStorage.__proto__.setItem = jest.fn();
+        const hook = await actAndFlush(() => renderHook(() => useAlertList(), { wrapper: MockProvider }));
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        expect(window.localStorage.getItem).toHaveBeenLastCalledWith('aiera:sdk:alertList');
+        expect(hook?.result.current.alertList).toEqual(initialAlertList);
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        window.localStorage.__proto__.getItem = jest.fn(() =>
+            JSON.stringify({
+                ...initialAlertList,
+                dateKeys: ['2022-01-04T16:30:00.000-05:00', '2022-01-05T05:59:00.000-05:00'],
+                '2022-01-05T05:59:00.000-05:00': ['2015016'],
+            })
+        );
+        await actAndFlush(() => {
+            hook?.result.current.addAlert('2022-01-05T05:59:00.000-05:00', '2015016');
+        });
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        expect(window.localStorage.setItem).toHaveBeenLastCalledWith(
+            'aiera:sdk:alertList',
+            JSON.stringify({
+                dateKeys: ['2022-01-04T16:30:00.000-05:00', '2022-01-05T05:59:00.000-05:00'],
+                '2022-01-04T16:30:00.000-05:00': ['2003019'],
+                '2022-01-05T05:59:00.000-05:00': ['2015016'],
+            })
+        );
+        expect(hook?.result.current.alertList).toEqual({
+            ...initialAlertList,
+            dateKeys: ['2022-01-04T16:30:00.000-05:00', '2022-01-05T05:59:00.000-05:00'],
```

Comment:
Is this right? Two date keys but only one mapped to the event id?

Reply:
in this example, initialAlertList had another mapped to an eventId